File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/reminders/RemindersListRequest.java
package com.github.seratch.jslack.api.methods.request.reminders;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

public class RemindersListRequest implements SlackApiRequest {

    /**
     * Authentication token. Requires scope: `reminders:read`
     */
    private String token;

    RemindersListRequest(String token) {
        this.token = token;
    }

    public static RemindersListRequestBuilder builder() {
        return new RemindersListRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public static class RemindersListRequestBuilder {
        private String token;

        RemindersListRequestBuilder() {
        }

        public RemindersListRequest.RemindersListRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public RemindersListRequest build() {
            return new RemindersListRequest(token);
        }

        public String toString() {
            return "RemindersListRequest.RemindersListRequestBuilder(token=" + this.token + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/reminders/RemindersListRequest.kt
package work.socialhub.kslack.api.methods.request.reminders

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest
import kotlin.js.JsExport

@JsExport
class RemindersListRequest(
    /** Authentication token. Requires scope: `reminders:read` */
    override var token: String?
) : SlackApiRequest, FormRequest {

    override fun toMap(): Map<String, Any> {
        return mapOf()
    }
}